### PR TITLE
Update winston definitions for v2.2.x

### PR DESCRIPTION
**Pull request by cjbarth**
case 2. Improvement to existing type definition.
- documentation or source code reference which provides context for the suggested changes.  url http://api.jquery.com/html .
  - it has been reviewed by a DefinitelyTyped member.

https://github.com/winstonjs/winston/blob/master/lib/winston/exception.js - Exception stuff
https://github.com/winstonjs/winston/blob/master/lib/winston.js - addRewriter has been removed

### CODE REVIEW

**Comment by cjbarth** [comment]
This depends on PR #8450 to pass.

File: winston/winston-tests.ts [hunk 1]
```diff
@@ -158,9 +158,9 @@ profiler.start = new Date();
 let testRewriter : winston.MetadataRewriter;
 testRewriter = function(level: string, msg: string, meta: any) {
     return meta;
-}
+};
 
-winston.addRewriter(testRewriter);
+logger.rewriters.push(testRewriter);
```
Comment (cjbarth) [inline winston/winston-tests.ts]: https://github.com/winstonjs/winston/blob/master/lib/winston.js - addRewriter as been removed

File: winston/winston.d.ts [hunk 2]
```diff
@@ -46,17 +52,65 @@ declare module "winston" {
...
+  export interface MetadataFilter {
+    (level: string, msg: string, meta: any): string | {msg: any; meta: any;};
+  }
+
   export interface LoggerStatic {
     new (options?: LoggerOptions): LoggerInstance;
   }
 
   export interface LoggerInstance extends NodeJS.EventEmitter {
+    rewriters: Array<MetadataRewriter>;
```
Comment (cjbarth) [inline winston/winston.d.ts]: https://github.com/winstonjs/winston/blob/master/lib/winston.js - rewriters replaces addRewriter

File: winston/winston.d.ts [hunk 3]
```diff
@@ -46,17 +52,65 @@ declare module "winston" {
...
+    filters: Array<MetadataFilter>;
```
Comment (cjbarth) [inline winston/winston.d.ts]: https://github.com/winstonjs/winston/blob/master/lib/winston.js

File: winston/winston.d.ts [hunk 4]
```diff
@@ -46,17 +52,65 @@ declare module "winston" {
...
+	transports: Array<TransportInstance>;
```
Comment (cjbarth) [inline winston/winston.d.ts]: https://github.com/winstonjs/winston/blob/master/lib/winston.js

File: winston/winston.d.ts [hunk 5]
```diff
@@ -129,6 +182,7 @@ declare module "winston" {
 
   export interface FileTransportInstance extends TransportInstance {
     new (options?: FileTransportOptions): FileTransportInstance;
+    close(): void;
```
Comment (cjbarth) [inline winston/winston.d.ts]: https://github.com/winstonjs/winston/blob/master/lib/winston/transports/file.js

File: winston/winston.d.ts [hunk 6]
```diff
@@ -269,10 +323,7 @@ declare module "winston" {
     start?: number;
     from?: Date;
     until?: Date;
-    /**
-     * 'asc' or 'desc'
-     */
-    order?: string;
+    order?: "asc" | "desc";
```
Comment (cjbarth) [inline winston/winston.d.ts]: https://github.com/winstonjs/winston/blob/master/lib/winston/transports/transport.js#L87

File: winston/winston.d.ts [hunk 7]
```diff
@@ -81,7 +135,6 @@ declare module "winston" {
     handleExceptions(...transports: TransportInstance[]): void;
     unhandleExceptions(...transports: TransportInstance[]): void;
     add(transport: TransportInstance, options?: TransportOptions, created?: boolean): LoggerInstance;
-    addRewriter(rewriter: MetadataRewriter): void;
```
Comment (cjbarth) [inline winston/winston.d.ts]: https://github.com/winstonjs/winston/blob/master/lib/winston.js#L68 - addRewriter has been removed

File: winston/winston.d.ts [hunk 8]
```diff
@@ -15,6 +19,8 @@ declare module "winston" {
   export var loggers: ContainerInstance;
   export var defaultLogger: LoggerInstance;
 
+  export var exception: Exception;
```
Comment (cjbarth) [inline winston/winston.d.ts]: https://github.com/winstonjs/winston/blob/master/lib/winston/exception.js#L12

File: winston/winston.d.ts [hunk 9]
```diff
@@ -46,17 +52,65 @@ declare module "winston" {
   export function addColors(target: any): any;
   export function setLevels(target: any): any;
   export function cli(): LoggerInstance;
-  export function addRewriter(rewriter: MetadataRewriter): void;
+  export function close(): void;
+
+  export interface ExceptionProcessInfo {
```
Comment (cjbarth) [inline winston/winston.d.ts]: https://github.com/winstonjs/winston/blob/master/lib/winston/exception.js#L24

File: winston/winston.d.ts [hunk 10]
```diff
@@ -46,17 +52,65 @@ declare module "winston" {
...
+    uid?: number;
+    gid?: number;
+    cwd: string;
+    execPath: string;
+    version: string;
+    argv: string;
+    memoryUsage: NodeJS.MemoryUsage;
+  }
+
+  export interface ExceptionOsInfo {
```
Comment (cjbarth) [inline winston/winston.d.ts]: https://github.com/winstonjs/winston/blob/master/lib/winston/exception.js#L37

File: winston/winston.d.ts [hunk 11]
```diff
@@ -46,17 +52,65 @@ declare module "winston" {
...
+    loadavg: [number, number, number];
+    uptime: number;
+  }
+
+  export interface ExceptionTrace {
```
Comment (cjbarth) [inline winston/winston.d.ts]: https://github.com/winstonjs/winston/blob/master/lib/winston/exception.js#L44

File: winston/winston.d.ts [hunk 12]
```diff
@@ -46,17 +52,65 @@ declare module "winston" {
...
+    column: number;
+    file: string;
+    "function": string;
+    line: number;
+    method: string;
+    native: boolean;
+  }
+
+  export interface ExceptionAllInfo {
```
Comment (cjbarth) [inline winston/winston.d.ts]: https://github.com/winstonjs/winston/blob/master/lib/winston/exception.js#L14

File: winston/winston.d.ts [hunk 13]
```diff
@@ -46,17 +52,65 @@ declare module "winston" {
...
+    date: Date;
+    process: ExceptionProcessInfo;
+    os: ExceptionOsInfo;
+    trace: Array<ExceptionTrace>;
+    stack: Array<string>;
+  }
+
+  export interface Exception {
```
Comment (cjbarth) [inline winston/winston.d.ts]: https://github.com/winstonjs/winston/blob/master/lib/winston/exception.js#L12

File: winston/winston.d.ts [hunk 14]
```diff
@@ -46,17 +52,65 @@ declare module "winston" {
...
+    getProcessInfo(): ExceptionProcessInfo;
+    getOsInfo(): ExceptionOsInfo;
+    getTrace(err: Error): Array<ExceptionTrace>;
+  }
 
   export interface MetadataRewriter {
     (level: string, msg: string, meta: any): any;
   }
 
+  export interface MetadataFilter {
```
Comment (cjbarth) [inline winston/winston.d.ts]: https://github.com/winstonjs/winston/blob/master/lib/winston/logger.js#L201

**Comment by vvakame** [comment]
I decided to merge this PR. thanks!

**Pull request merged**